packages/x-charts/src/RadarChart/RadarSeriesPlot/RadarSeriesPlot.tsx

```diff
@@ -28,21 +29,32 @@ function RadarSeriesPlot(props: RadarSeriesPlotProps) {
                 key={seriesId}
                 d={getAreaPath(points)}
                 {...interactionProps[seriesIndex]}
-                className={
-                  (isItemHighlighted && classes.highlighted) ||
-                  (isItemFaded && classes.faded) ||
-                  undefined
-                }
-                fill={color}
+                className={clsx(
+                  classes.area,
+                  (isItemHighlighted && classes.highlighted) || (isItemFaded && classes.faded),
+                )}
+                fill={fillArea ? color : 'transparent'}
+                fillOpacity={(isItemHighlighted && 0.4) || (isItemFaded && 0.1) || 0.2}
```

Comment:
Should we use a ternary instead? Probably more readable

Reply:
> Probably more readable

Personally, I found this logical writing more readable than nested ternary.

Reply:
I prefer this way as well. Otherwise I would go with a function instead of ternary